src/dbnode/storage/series/types.go

```diff
@@ -283,3 +329,42 @@ func NewStats(scope tally.Scope) Stats {
 func (s Stats) IncCreatedEncoders() {
 	s.encoderCreated.Inc(1)
 }
+
+// WriteType is an enum for warm/cold write types.
+type WriteType int
+
+const (
+	// UndefinedWriteType is an undefined write type.
+	UndefinedWriteType WriteType = iota
+
+	// WarmWrite represents warm writes (within the buffer past/future window).
+	WarmWrite
+
+	// ColdWrite represents cold writes (outside the buffer past/future window).
+	ColdWrite
+)
+
+// WriteOptions define different options for a write.
+type WriteOptions struct {
+	WriteType WriteType
+}
+
+// ResolveWriteType returns whether a write is a cold write or warm write.
+func (w *WriteOptions) ResolveWriteType(
```

Comment:
Can we make this method a NewWriteType function? Instead of a method on WriteOptions? Ideally, WriteOptions do not have much logic, just some defaulter/getter/setter.

Reply:
If you are close to land, it is ok not to make these changes in short notice, I would be happy to include the suggested change in my pr (that make use of the WriteOptions to pass down schema). Let me know which way you prefer.